apps/nextjs/src/components/board/sections/content.tsx

```diff
@@ -112,22 +115,51 @@ const BoardItemContent = ({ item, ...dimensions }: ItemContentProps) => {
   if (!serverData?.isReady) return null;
 
   return (
-    <>
-      <ItemMenu offset={4} item={newItem} />
-      <Comp
-        options={options as never}
-        integrations={item.integrations}
-        serverData={serverData?.data as never}
-        isEditMode={editMode}
-        boardId={board.id}
-        itemId={item.id}
-        {...dimensions}
-      />
-    </>
+    <QueryErrorResetBoundary>
+      {({ reset }) => (
+        <ErrorBoundary
+          onReset={reset}
+          fallbackRender={({ resetErrorBoundary, error }) => (
+            <>
+              <ItemMenu
+                offset={4}
+                item={newItem}
+                resetErrorBoundary={resetErrorBoundary}
+              />
+              <WidgetError
+                kind={item.kind}
+                error={error as unknown}
+                resetErrorBoundary={resetErrorBoundary}
+              />
+            </>
+          )}
+        >
+          <ItemMenu offset={4} item={newItem} />
+          <Comp
+            options={options as never}
+            integrationIds={item.integrationIds}
+            serverData={serverData?.data as never}
+            isEditMode={editMode}
+            boardId={board.id}
+            itemId={item.id}
+            {...dimensions}
+          />
+        </ErrorBoundary>
+      )}
+    </QueryErrorResetBoundary>
   );
 };
 
-const ItemMenu = ({ offset, item }: { offset: number; item: Item }) => {
+const ItemMenu = ({
+  offset,
+  item,
+  resetErrorBoundary,
+}: {
+  offset: number;
+  item: Item;
+  resetErrorBoundary?: () => void;
```

Comment:
Consider making this not optional

Reply:
Not possible as it's not possible to be defined outside of the error boundary